compiler/rustc_mir/src/transform/promote_consts.rs

```diff
@@ -60,6 +60,10 @@ impl<'tcx> MirPass<'tcx> for PromoteTemps<'tcx> {
             return;
         }
 
+        if !tcx.consider_optimizing(|| format!("PromoteTemps {:?} ", body.source.def_id())) {
```

Comment:
promotion is a guaranteed optimization... I guess it's not problematic to do this, as it's just a debugging tool, but it should happen in `collect_temps_and_candidates` for each candidate

Reply:
There are some other passes that I did not add checks to because they looked like they were not strictly "optimizations" (they added information, or they were lints, for example). Sounds like this file could be included in that group as well? If it's guaranteed, we should run it whether or not we have fuel left?

Reply:
ah, yea, while it would be practical to be able to bisect promotions sometimes, we'd require more fine grained control than the fuel. So yea, let's not do this for `promote_consts` at all